writer.go

```diff
@@ -54,14 +56,25 @@ func (entry *Entry) WriterLevel(level Level) *io.PipeWriter {
 	return writer
 }
 
+// writerScanner scans the input from the reader and writes it to the logger
 func (entry *Entry) writerScanner(reader *io.PipeReader, printFunc func(args ...interface{})) {
-	scanner := bufio.NewScanner(reader)
-	for scanner.Scan() {
-		printFunc(scanner.Text())
-	}
-	if err := scanner.Err(); err != nil {
-		entry.Errorf("Error while reading from Writer: %s", err)
+	for {
+		data := make([]byte, bufio.MaxScanTokenSize)
+		c, err := reader.Read(data)
+		if err != nil && err != io.EOF {
+			entry.Errorf("Error while reading from Writer: %s", err)
+			break
+		}
+
+		if c > 0 {
+			chunks := strings.Split(strings.TrimRight(string(bytes.Trim(data, "\x00")), "\n"), "\n")
```

Comment:
why is the zero byte trimmed, this feels arbitrary?

Reply:
During testing, for data that was < 64KB, the remaining bytes were `\x00` which causes issues, so `bytes.Trim(data, "\x00")` was there to clean this up. I've since improved the logic to only process the number of read bytes returned by `reader.Read()` which improves things, WDYT?